src/endpoints/Branches.js

```diff
@@ -58,17 +59,35 @@ export default class Branches extends Endpoint {
 
     return this.configureRequest<Promise<Branch[]>>("list", {
       api: async () => {
-        const query = querystring.stringify({ limit, offset, filter, search });
-        const requestUrl = descriptor
-          ? `projects/${descriptor.projectId}/branches/?${query}`
-          : `branches/?${query}`;
+        const queryOptions = { limit, offset, filter, search };
+        let query = querystring.stringify(queryOptions);
+        let response = null;
+
+        if (descriptor && descriptor.userId) {
+          query = querystring.stringify({
+            ...queryOptions,
+            userId: descriptor.userId
+          });
+
+          response = await this.apiRequest(`branches/?${query}`, {
+            headers
+          });
+        } else if (descriptor && descriptor.projectId) {
+          response = await this.apiRequest(
+            `projects/${descriptor.projectId}/branches/?${query}`,
+            { headers }
+          );
+        } else {
+          response = await this.apiRequest(`branches/?${query}`, {
+            headers
+          });
+        }
```

Comment:
```suggestion
        if (descriptor && descriptor.userId) {
          query = querystring.stringify({
            ...queryOptions,
            userId: descriptor.userId
          });
        }

        if (descriptor && descriptor.projectId) {
          response = await this.apiRequest(
            `projects/${descriptor.projectId}/branches/?${query}`,
            { headers }
          );
        } else {
          response = await this.apiRequest(`branches/?${query}`, {
            headers
          });
        }
```

Bonus: This lets you search for a users branches within a project?